File: android/src/main/java/com/easemob/im_flutter_sdk/EMProgressManager.java
package com.easemob.im_flutter_sdk;

import com.hyphenate.EMError;
import com.hyphenate.exceptions.HyphenateException;

import java.util.HashMap;
import java.util.Map;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.JSONMethodCodec;
import io.flutter.plugin.common.MethodChannel;

public class EMProgressManager extends EMWrapper implements MethodChannel.MethodCallHandler {

    public EMProgressManager(FlutterPlugin.FlutterPluginBinding flutterPluginBinding, String channelName) {
        super(flutterPluginBinding, channelName);
    }

    public void sendDownloadProgressToFlutter(String fileId, int progress){
        HashMap<String, Object> data = new HashMap<>();
        data.put("fileId", fileId);
        data.put("progress", progress);
        post(()->{
            channel.invokeMethod("onProgress", data);
        });
    }

    public void sendDownloadSuccessToFlutter(String fileId, String savePath) {
        HashMap<String, Object> data = new HashMap<>();
        data.put("fileId", fileId);
        data.put("savePath", savePath);
        post(()->{
            channel.invokeMethod("onSuccess", data);
        });
    }

    public void sendDownloadErrorToFlutter(String fileId, HyphenateException e) {
        Map<String, Object> error = new HashMap<>();
        error.put("code", e.getErrorCode());
        error.put("description", e.getDescription());
        HashMap<String, Object> data = new HashMap<>();
        data.put("fileId", fileId);
        data.put("error", error);
        post(()->{
            channel.invokeMethod("onError", data);
        });
    }
}

File: ios/Classes/EMProgressManager.m
//
//  EMProgressManager.m
//  im_flutter_sdk
//
//  Created by 杜洁鹏 on 2022/5/23.
//
#import <Flutter/Flutter.h>
#import <HyphenateChat/HyphenateChat.h>
#import "EMProgressManager.h"
#import "EMError+Helper.h"

@interface EMProgressManager ()

@end


@implementation EMProgressManager

- (instancetype)initWithChannelName:(NSString *)aChannelName
                          registrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    if(self = [super initWithChannelName:aChannelName
                               registrar:registrar]) {

    }
    return self;
}

- (void)sendDownloadSuccessToFlutter:(NSString *)fileId path:(NSString *)savePath {
    [self.channel invokeMethod:@"onSuccess" arguments:@{@"fileId":fileId, @"savePath": savePath}];
}

- (void)sendDownloadProgressToFlutter:(NSString *)fileId progress:(int)progress {
    [self.channel invokeMethod:@"onProgress" arguments:@{@"fileId":fileId, @"progress": @(progress)}];
}

- (void)sendDownloadErrorToFlutter:(NSString *)fileId error:(EMError *)error {
    [self.channel invokeMethod:@"onError" arguments:@{@"fileId":fileId, @"error": [error toJson]}];
}


- (void)unRegisterEaseListener {
}


@end
